pkg/operator/controller/externaldns/pod.go

```diff
@@ -170,7 +171,58 @@ func (b *externalDNSContainerBuilder) fillProviderAgnosticFields(seq int, zone s
 
 	//TODO: Add logic for the CRD source.
 
+	filterArgs, err := b.domainFilters()
+	if err != nil {
+		return err
+	}
+	container.Args = append(container.Args, filterArgs...)
 	container.Args = append(container.Args, args...)
+	return nil
+}
+
+func (b *externalDNSContainerBuilder) domainFilters() ([]string, error) {
+	var (
+		regexDomainIncludeSet bool
+		regexDomainExcludeSet bool
+		args                  []string
+	)
+	for _, d := range b.externalDNS.Spec.Domains {
+		switch d.FilterType {
+		case operatorv1alpha1.FilterTypeInclude:
+			switch d.MatchType {
+			case operatorv1alpha1.DomainMatchTypeExact:
+				if d.Name == nil {
+					return nil, fmt.Errorf("name for domain cannot be empty")
+				}
```

Comment:
In the future, this validation could be handled by an admission webhook (which could also validate that `d.Name` *is* nil when `d.MatchType` is *not* `operatorv1alpha1.DomainMatchTypeExact`).  Likewise for the regex case.

Reply:
I wanted to do most of the validation in a webhook but we were not sure if that would be ok with netedge.